compiler/rustc_span/src/lib.rs

```diff
@@ -1333,7 +1352,15 @@ impl<D: SpanDecoder> Decodable<D> for Span {
 
 impl<D: SpanDecoder> Decodable<D> for Symbol {
     fn decode(s: &mut D) -> Symbol {
-        s.decode_symbol()
+        s.decode_symbol_index_or_byte_str(Symbol::new, |b| {
+            Symbol::intern(unsafe { str::from_utf8_unchecked(b) })
```

Comment:
Hmm, is this unsafe justified?
We are reading data from some file (?) and just assume that it's UTF-8.

Seems pre-existing because `Decoder::read_str` does that as well, but it at least does a `STR_SENTINEL` check.